mitodl/open-discussions
# updates indexing code for course catalog models

_This is PR is intended to start the necessary conversation about the exact behavior we want from the indexing code of the new learning resource objects._

#### What are the relevant tickets?
Closes #2042

#### What's this PR do?
Updates indexing-related code for the new learning resource models, namely for Bootcamps. In addition to indexing related code, this PR adds analog functions for creating, updating, and deleting associated index objects for Bootcamps, Programs, and UserLists, where we are creating, updating, deleting Bootcamps, Programs, and UserLists; however, they are not used any where yet. There are no Programs of UserLists currently. There do exists 3 Bootcamps.

#### How should this be manually tested?
run ES indexing code, check for the created ES indices, search for those objects
1) Create index:
```
docker-compose  run web python manage.py recreate_index
```
2) Hit this endpoint to run a search query for bootcamp: `http://localhost:9101/discussions_local_all_default/_search`
with a POST request:
Method: `POST`
Content-Type: `application/json`
Body: `{"query":{"term":{"object_type":"bootcamp"}}}`
